models/articles.model.js

```diff
@@ -61,9 +61,19 @@ exports.selectAllArticles = (query) => {
   }
   queryString += ` 
   GROUP BY articles.article_id 
-  ORDER BY articles.created_at DESC;`;
+  ORDER BY ${sort_by} ${order};`;
 
   return db.query(queryString, queryValues).then(({ rows }) => {
+    if (!rows.length && topic) {
+      return db
+        .query(`SELECT * FROM topics WHERE slug = $1;`, [topic])
+        .then(({ rows: topicRows }) => {
+          if (!topicRows.length) {
+            return Promise.reject({ status: 404, msg: "Topic not found" });
+          }
+          return [];
+        });
+    }
```

Comment:
Nice logic, but this current has to wait for the first query to happen before we can check this. Any ideas how we could do both of these at the same time so we dont have to wait for one to finish before trying the other?